frontend/src/api/task.ts

```diff
@@ -0,0 +1,68 @@
+import { post } from "src/api/requests";
+
+/**
+ * Initializes the shape for the data being retrieved from the backend.
+ */
+export interface Task {
+  _id: string;
+  title: string;
+  description?: string;
+  isChecked: boolean;
+  dateCreated: Date;
+}
+
+interface TaskJSON {
+  _id: string;
+  title: string;
+  description?: string;
+  isChecked: boolean;
+  dateCreated: string;
+}
+
+/**
+ * Converts taks from "any"-typed JSON that only contains primitive types to our
+ * custom Task interface. This is used to change the date field from a string
+ * to a Date object.
+ *
+ * @param task the JSON representation of the task
+ * @returns the typed Task
+ */
+function parseTask(task: TaskJSON): Task {
+  return {
+    _id: task._id,
+    title: task.title,
+    description: task.description,
+    isChecked: task.isChecked,
+
+    // convert the string version of the date to a Date object
+    dateCreated: new Date(task.dateCreated),
+  };
+}
+
+/**
+ * This interface is optional, it's being used to make the input of createTask
+ * cleaner. It's being exported so when calls are made to the function it can
+ * be used.
+ */
+export interface CreateTaskRequest {
```

Comment:
Calling this interface "optional" might be a little confusing - I assume you mean that it's not required to define a type like this in order to make an HTTP request, but neither is it required to write any of these helper functions so the point is kind of moot.

```suggestion
/**
 * The expected inputs when we want to create a new Task object. In the MVP, we only
 * need to provide the title and optionally the description, but in the course of
 * this tutorial you'll likely want to add more fields here.
 */
export interface CreateTaskRequest {
```